helix-term/src/commands.rs

```diff
@@ -679,7 +679,20 @@ fn kill_to_line_start(cx: &mut Context) {
 
     let selection = doc.selection(view.id).clone().transform(|range| {
         let line = range.cursor_line(text);
-        range.put_cursor(text, text.line_to_char(line), true)
+        let first_char = text.line_to_char(line);
+        let anchor = range.cursor(text);
+        let head = if anchor == first_char && line != 0 {
+            line_end_char_index(&text, line - 1)
+        } else if let Some(pos) = find_first_non_whitespace_char(text.line(line)) {
+            if first_char + pos < anchor {
+                first_char + pos
+            } else {
+                first_char
+            }
+        } else {
+            first_char
+        };
```

Comment:
This whole block has no comments and is kind of hard for me to understand what it's doing.

Reply:
Sorry, I thought it was clear enough, added comments now.